src/handlers.cpp

```diff
@@ -162,6 +162,15 @@ void ImpiTask::run()
   }
 }
 
+ImpiTask::~ImpiTask()
+{
+  if (_maa != NULL)
```

Comment:
There's no need to check `if (_maa != NULL)` - `delete _maa` implicitly checks that _maa is non-NULL.